Source/Plugins/Core/com.equella.core/src/com/tle/integration/lti/canvasextension/CanvasLtiWrapperExtension.java

```diff
@@ -37,11 +38,29 @@ public String getUserId(HttpServletRequest request) {
     return request.getParameter("custom_canvas_user_id");
   }
 
+  @Override
```

Comment:
I'd like to ignore the custom parms for Canvas and D2L lti signons, with the hope to bring all LTI signons over to the generic flow eventually.

Reply:
Canvas wrapper would be easy to change, but the rest of the integration has a lot of specific code and parameters, so the only way to make it generic would be to add all that specific code as "if canvas" sections in the generic code.  Do we want to "mix" the canvas code with the generic code in only one code, or do we prefer to keep it "isolated". Same with D2L

Reply:
I'm thinking something like the following - just completely ignore for non-generic LTI wrappers:
```
  public String getUserId(HttpServletRequest request, String param) {
    return getUserId(request);
  }
```